astropy/convolution/convolve.py

```diff
@@ -20,24 +21,19 @@
 from ..modeling.core import _CompoundModelMeta
 from .utils import KernelSizeError, has_even_axis, raise_even_kernel_exception
 
-# Turn the faulthandler ON to help catch any signals, e.g. segfaults
-# or asserts. This doesn't, currently, work with Jupyter Notebook.
-faulthandler.enable()
-
-# Find and load C convolution library
-lib_path = glob.glob(os.path.join(os.path.dirname(__file__), 'lib_convolve*'))[0]
-if sys.platform.startswith('win'):
-    libConvolve = ctypes.windll.LoadLibrary(lib_path)
-else:
-    libConvolve = ctypes.cdll.LoadLibrary(lib_path)
+LIBRARY_PATH = os.path.dirname(__file__)
 
+try:
+    lib_convolve = load_library("lib_convolve", LIBRARY_PATH)
+except Exception:
+    raise Exception("Convolution C extension is missing. Try re-building astropy.")
```

Comment:
@astrofrog - I'm sorry I haven't realized this earlier, but isn't `Exception` a bit too generic here? Wouldn't be e.g. `ImportError` be better?

Reply:
I'm not sure what the full range of exceptions is, but to be honest if *any* exception happens here I think we want to show a nice message for users. I think it could raise OSError and probably some other kinds.

Reply:
Oh right duh. Yes this should raise an ImportError. Maybe @jamienoss can include that in the PR he is working on?

Reply:
We should just include https://github.com/astropy/astropy/pull/8076 in 3.1